# Author: Стурейко Игорь
## Project: Geekbrains.MatStat
## Lesson 01 - Случайные события. Условная вероятность. Формула Байеса. Независимые испытания
## Date: 2020-06-22

## 1. Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

1. Посчитаем скольктми способами можно вынуть тузы из 4: $C_4^1=4, C_4^2=6, C_4^3=4, C_4^4=1$

2. Тогда итоговое число вариантов: $C_4^1 \cdot C_{48}^3 + C_4^2 \cdot C_{48}^2 + C_4^3 \cdot C_{48}^1 + C_4^4$ 

## 2. Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что рядом будут сидеть: а) два определённых человека? б) три определённых человека?

a) Сначала сколькими способами можно взять 2 определенных человек из 7: $C_7^2$, затем нам нужно чтобы они сидели рядом - т.е это получается 6 вариантов из всех возможных: $\displaystyle P(2) = \frac{6}{C_7^2}=\frac{6}{21}=\frac{2}{7}$

b) 3 определенных человека это $C_7^3=35$, вариантов когда они сидят вместе - 5, таким образом: $\displaystyle P=\frac{5}{C_7^3}=\frac{5}{35}=\frac{1}{7}$

In [1]:
import numpy as np
def combinations(n: int, k: int) -> int:
    """Число сочетаний.
    """
    
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n - k))

In [5]:
combinations(60, 3)

34220

## 3. Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

Аналогично 1. Количество вариантов выбрать 3 вопроса из 50 тех которые он знает $C_{50}^3$, и количество вариантов вообще выбрать 3 вопроса $C_{60}^3$, тогда итоговая вероятность отношение числа исходов: $\displaystyle P=\frac{C_{50}^3}{C_{60}^3}=\frac{19600}{34220}=\frac{4900}{8555}$

## 4. Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

Вероятность $\displaystyle P(A)=\frac{1}{2}, P(B)=\frac{1}{2}$, тогда $\displaystyle P(A|B)=\frac{P(AB)}{P(B)}=\frac{1/4}{1/2}=\frac{1}{2}=P(A) \to$ события независимы

## 5. Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ. Вопрос: какова вероятность, что вы действительно больны ей? Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.